prog/prio.go

```diff
@@ -48,36 +48,30 @@ func (target *Target) calcStaticPriorities() [][]float32 {
 					// Self-priority is assigned below.
 					continue
 				}
-				prios[c0][c1] += w0 * w1
+				// The static priority is assigned based on the direction of arguments. A higher priority will be
+				// assigned when c0 is a call that produces a resource and c1 a call that uses that resource.
+				prios[c0][c1] += w0.inout*w1.in + 0.7*w0.inout*w1.inout
 			}
 		}
 	}
-
-	// Self-priority (call wrt itself) is assigned to the maximum priority
-	// this call has wrt other calls. This way the priority is high, but not too high.
+	// The value assigned for self-priority (call wrt itself) have to be high, but not too high (1.0).
```

Comment:
Move this one line down.
Also drop the "(1.0)" part. It's now not 1.0 and it's now also obvious what's the resulting self priority.